apps/spruce/src/pages/task/taskTabs/TaskHistory/TaskTimeline/DateSeparator/index.tsx

```diff
@@ -0,0 +1,63 @@
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { palette } from "@leafygreen-ui/palette";
+import { size } from "@evg-ui/lib/constants/tokens";
+
+const { gray } = palette;
+
+export const DATE_SEPARATOR_WIDTH = 8;
+interface DateSeparatorProps {
+  date?: Date | null;
+}
+const DateSeparator: React.FC<DateSeparatorProps> = ({ date }) => {
+  const parsedDate = new Date(date || "");
+  const formattedDate = parsedDate?.toLocaleDateString("en-US", {
+    month: "numeric",
+    day: "2-digit",
+  });
+  const dateString = formattedDate || "";
+  return (
+    <Container aria-label="date-separator" className="date-separator">
+      <StyledBadge className="date-badge">{dateString}</StyledBadge>
+      <Dot className="dot" />
+      <Line />
+    </Container>
+  );
+};
+
+const Container = styled.div`
+  width: ${DATE_SEPARATOR_WIDTH}px;
+  height: ${size.s};
+  position: relative;
+  cursor: pointer;
+`;
+const StyledBadge = styled(Badge)`
+  position: absolute;
+  top: -${size.m};
+  left: 50%;
+  transform: translateX(-50%);
+  background-color: white;
+  border-radius: ${size.m};
+  padding: ${size.xxs};
+`;
+
+const Dot = styled.div`
+  width: ${size.xxs};
+  height: ${size.xxs};
+  background-color: ${gray.light1};
+  border-radius: 50%;
+  position: absolute;
+  top: -${size.s};
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+const Line = styled.div`
```

Comment:
What do you think about dashed lines? In a parallel to the waterfall?
<img width="719" alt="Screenshot 2025-04-30 at 3 18 49 PM" src="https://github.com/user-attachments/assets/07df2133-b517-4069-9e0e-689995c514b6" />